interface/admin.py

```diff
@@ -45,6 +50,26 @@ def download_submissions(self, request, queryset):
 
     download_submissions.short_description = 'Download submissions for review'
 
+    def download_all_submissions(self, request, queryset):
+        submission_set = self.get_submission_set(request, queryset)
+
+        with TemporaryDirectory() as _tmp:
+            tmp = Path(_tmp)
+
+            with ZipFile(tmp / 'review.zip', 'x') as zipfile:
+                for submission in submission_set:
+                    submission.download(tmp / f'{submission.id}.zip')
+                    zipfile.write(
+                        tmp / f'{submission.id}.zip',
+                        f'{submission.user.username}-{submission.id}.zip',
+                    )
+
+            review_zip = (tmp / 'review.zip').open('rb')
```

Comment:
This code overlaps with the one above. This bit should be moved in a separate function, something like `archive_submissions(submissions)` that will return a `File object` (i.e. `review_zip`).